### PR TITLE
Features+fixes/zl/cuttlefishisms+fuse handling around ed

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
- Adds fuse/batch settings to cuttlefish w/ a few validations
- More types... one TODO for @fadushin as well!
- Don't attempt to build tree if Indexes are blowing during folding of keys for a partition. 
- This also includes the logging fix which is part of #565 now. 
- Cherry-picked @jonmeredith's addition/patch to our hotfix branch -> Make rpc timeout for SOLR port configurable.

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
@fadushin @jvoegele I had issues w/ yz_aae_test running this. It was flaky. Running these changes atop #565 always passed. So, it doesn't seem like these changes affect it.

File: src/yz_solrq.erl [hunk 1]
```diff
@@ -70,7 +74,7 @@
     }
 ).
 
-
+%% TODO: Clean this up w/ proper type naming conventions.
```
Comment (zeeshanlakhani) [inline src/yz_solrq.erl:77]: @fadushin or @jvoegele would love to remove this on main branch or w/e works best.

File: priv/yokozuna.schema [hunk 2]
```diff
@@ -101,3 +101,132 @@
...
+  {default, on},
+  {datatype, flag},
+  hidden
+]}.
+
+%%===========================================================
+%% Batching-related mappings for SolrQ workers
+%%===========================================================
+
+%% @doc The minimum batch size, in number of Riak objects. Any batches that
```
Comment (zeeshanlakhani) [inline priv/yokozuna.schema:152]: @fadushin your batch doc came in super helpful! :)

File: priv/yokozuna.schema [hunk 3]
```diff
@@ -101,3 +101,132 @@
...
+%%      are smaller than this amount will not be immediately flushed to Solr,
+%%      but are guaranteed to be flushed within solrq_delayms_max milliseconds.
+{mapping, "search.solrq_batch_min", "yokozuna.solrq_batch_min", [
+  {default, 1},
+  {datatype, integer},
+  hidden,
+  {validators, ["positive_integer"]}
+]}.
+
+%% @doc The maximim batch size, in number of Riak objects. Any batches that are
```
Comment (kesslerm) [inline priv/yokozuna.schema]: s/maximim/maximum/g

File: priv/yokozuna.schema [hunk 4]
```diff
@@ -101,3 +101,132 @@
...
+%%      if writes into Solr start to fall behind.
+{mapping, "search.solrq_queue_hwm", "yokozuna.solrq_queue_hwm", [
+  {default, 10000},
+  {datatype, integer},
+  hidden,
+  {validators, ["integer"]}
+]}.
+
+%% @doc The number of solr queue workers to instantiate in the Yokozuna
+%%      application. Solr queue workers are responsible for enqueing objects for
```
Comment (kesslerm) [inline priv/yokozuna.schema]: s/enqueing/enqueueing ?

**Comment by zeeshanlakhani** [comment]
Will be integrated into other PR.

**Pull request closed without merging**